verde/base/base_classes.py

```diff
@@ -559,3 +571,30 @@ def get_instance_region(instance, region):
             raise ValueError("No default region found. Argument must be supplied.")
         region = getattr(instance, "region_")
     return region
+
+
+def get_extra_coords_names(coordinates, basename):
```

Comment:
I would make this a method (maybe private for now) so you don't have to pass in `self.extra_coords_name` and it can be modified by derived classes.